### PR TITLE
Design Membership list page

### LINKED ISSUE #12: Design Registrations List view component

### PR DESCRIPTION
**Pull request by anthonyminyungi**
- It closes #12 
- This PR contains implementation of `fullScreen`.

### CODE REVIEW

File: src/components/common/info/index.tsx [hunk 1]
```diff
@@ -13,23 +14,24 @@ const useStyles = makeStyles(() => ({
...
+export default function WebsiteInfo({ title, favicon, centered }: WebsiteInfoProps) {
     const classes = useStyles();
+    
 
     return (
-        <Container className={classes.container}>
+        <Container className={classes.container} style={ centered ? { justifyContent: "center" } : {}}>
             <Avatar className={classes.icon} alt={title} src={favicon} />
-            <Typography className={classes.title}>
+            <Typography className={classes.title} noWrap>
```
Comment (anthonyminyungi) [inline src/components/common/info/index.tsx:34]: Fixed because of the usage in different way.

File: src/components/current/index.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import {
 
 type CurrentInfo = {
     label: string,
-    data: string,
+    data: string | number,
```
Comment (anthonyminyungi) [inline src/components/current/index.tsx:12]: The data could be provided as a number.

File: src/components/main/index.tsx [hunk 3]
```diff
@@ -94,10 +116,140 @@ function a11yProps(index: number) {
...
+            {
+                label: 'email',
+                data: 'abc@abcd.com',
+            },
+            {
+                label: 'birth',
+                data: '2020-01-01',
+            },
+        ],
+    },
```
Comment (anthonyminyungi) [inline src/components/main/index.tsx:168]: Dummy data structure for the membership list.

File: src/components/main/index.tsx [hunk 4]
```diff
@@ -94,10 +116,140 @@ function a11yProps(index: number) {
...
+    return (
+      <MuiDialogTitle disableTypography className={classes.root}>
+        {onClose ? (
+          <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
+            <CloseIcon />
+          </IconButton>
+        ) : null}
+      </MuiDialogTitle>
+    );
+  });
```
Comment (anthonyminyungi) [inline src/components/main/index.tsx:136]: For the close button in the `fullScreen` dialog.

File: src/components/main/index.tsx [hunk 5]
```diff
@@ -121,10 +273,14 @@ export default function MainPage({ title, favicon, currentInfos, signature }: Ma
...
-                    <Button className={classes.seeMore} startIcon={<FullscreenIcon />}>
+                    <Button className={classes.seeMore} startIcon={<FullscreenIcon />} onClick={handleClickOpen}>
                         Full Screen
                     </Button>
-                    <MembershipListPage />
+                    <Dialog open={open} fullScreen onClose={handleClose} TransitionComponent={Transition}>
+                        <DialogHeader onClose={handleClose} />
+                        <MembershipListPage memberships={dummyMemberships} />    
+                    </Dialog>
+                    <MembershipListPage memberships={dummyMemberships} />
```
Comment (anthonyminyungi) [inline src/components/main/index.tsx:283]: Appearance.
Needed to check if it also shows a browser-sized full screen or not.
스크린샷 2020-09-02 오후 5 48 29

File: src/components/membership/index.tsx [hunk 6]
```diff
@@ -1,15 +1,174 @@
...
+                    memberships.map((membership, index) => {
+                        const id = `${index}-${membership.title}`;
+                        return (
+                            <span key={membership.title}>
+                                <ListItem className={classes.listItem} button divider onClick={() => handleClickOpen(id)}>
+                                    <WebsiteInfo title={membership.title} favicon={membership.favicon} />
+                                    <ListItemIcon>
+                                        <ChevronRightIcon />
+                                    </ListItemIcon>
+                                </ListItem>
```
Comment (anthonyminyungi) [inline src/components/membership/index.tsx:106]: Appearance.
스크린샷 2020-09-02 오후 5 49 49

File: src/components/membership/index.tsx [hunk 7]
```diff
@@ -1,15 +1,174 @@
...
+                                    <DialogContent>
+                                        {
+                                            membership.infos.map((i) => {
+                                                return (
+                                                    <Typography key={i.label} className={classes.text}>
+                                                        {i.label.toLowerCase()} : {i.data}
+                                                    </Typography>
+                                                )
+                                            })
+                                        }
```
Comment (anthonyminyungi) [inline src/components/membership/index.tsx:129]: Appearance when the element of the list clicked (modal opened).
스크린샷 2020-09-02 오후 5 51 00

**Review by anthonyminyungi — COMMENTED** [review]
Self-review

**Review by HyeockJinKim — APPROVED** [review]
LGTM

**Pull request merged**